app/lib/authentication/strategy/cas.rb

```diff
@@ -51,7 +51,7 @@ def login_url
       end
 
       def login_url_with_service
-        login_url + "?service=" + URI.escape(service)
+        login_url + "?service=" + CGI.escape(service)
```

Comment:
ah, we had this discussion in the past but I forgot :(

Was this the proper way to escape a single query parameter value?

Reply:
Let me paste a link to a random internet user comment: https://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape#comment137503555_8756159

perhaps do like 
```suggestion
        login_url + "?" + {service: service}.to_param
```

Reply:
I think I updated these invocations so that the tests still pass :grimacing: 
This specific method though seems to be already obsolete and unused. We can probably get rid of it in another PR.

I've now added a test, and your updated code seems to be working the same way, so your suggestion is fine for me.